core/kernel/src/feature_manifest.rs

```diff
@@ -0,0 +1,270 @@
+                        .author
+                        .clone()
+                        .or(full_manifest.author),
+                    version: translation
+                        .version
+                        .clone()
+                        .or(full_manifest.version),
+                    description: translation
+                        .description
+                        .clone()
+                        .or(full_manifest.description),
+                    thumbnail: translation
+                        .thumbnail
+                        .clone()
+                        .or(full_manifest.thumbnail),
+                    thumbnail_color: translation
+                        .thumbnail_color
+                        .clone()
+                        .or(full_manifest.thumbnail_color),
+                    primary_color: translation
+                        .primary_color
+                        .clone()
+                        .or(full_manifest.primary_color),
+                    links: Some(links),
+                }
+            }
+            None => FeatureManifest {
+                name: full_manifest.name,
+                author: full_manifest.author,
+                version: full_manifest.version,
+                description: full_manifest.description,
+                thumbnail: full_manifest.thumbnail,
+                thumbnail_color: full_manifest.thumbnail_color,
+                primary_color: full_manifest.primary_color,
+                links: full_manifest.links,
+            },
+        }
+    }
+
+    fn default() -> FeatureManifest {
+        FeatureManifest {
+            name: None,
+            author: None,
+            version: None,
+            description: None,
+            thumbnail: None,
+            thumbnail_color: None,
+            primary_color: None,
+            links: None,
+        }
+    }
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct FullFeatureManifest {
+    name: Option<String>,
+    author: Option<String>,
+    version: Option<String>,
+    description: Option<String>,
+    thumbnail: Option<String>,
+    thumbnail_color: Option<String>,
+    primary_color: Option<String>,
+    links: Option<HashMap<String, String>>,
+    translations: Option<HashMap<String, FeatureManifest>>,
+}
+
+impl TryFrom<&JSONStr> for FullFeatureManifest {
+    type Error = FeatureManifestParsingError;
+
+    fn try_from(value: &JSONStr) -> Result<Self, Self::Error> {
+        serde_json::from_str(value).map_err(|err| FeatureManifestParsingError {
+            description: err.to_string(), 
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_empty_json() {
+        let parsed = FeatureManifest::parse("", "");
+
+        assert!(parsed.is_err())
+    }
+
+    #[test]
+    fn test_empty_json_object() {
+        let parsed = FeatureManifest::parse("{}", "");
+
+        assert_eq!(parsed.unwrap(), FeatureManifest::default())
```

Comment:
Is this where you're using `PartialEq`? Or macros don't require it?

Reply:
Still, adding code just for testing... I don't know. Is there an alternative?

Reply:
I guess comparing fields by hand, I don't see a big issue with having "PartialEq" derived. I can remove it and have by fields comparison, but I also expect in the future to have additional equality checks in tests - for example when we would allow kernel to load a list of FeatureManifest.

Reply:
I'm OK with whatever you decide, including leaving it as it is now.